src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+                    verWeCanReadBack: 0x00010001,
+                    loaderSignature: LoaderSignature,
+                    loaderAssemblyName: typeof(SrCnnEntireModeler).Assembly.FullName);
+            }
+
+            public SrCnnEntireModeler(IHostEnvironment env, SrCnnEntireTransformer parent, IDataView data)
+                : this(env, parent)
+            {
+                Train(data);
+            }
+
+            private SrCnnEntireModeler(IHostEnvironment env, SrCnnEntireTransformer parent)
+            {
+                Contracts.CheckValue(env, nameof(env));
+                Contracts.CheckValue(parent, nameof(parent));
+                _host = env.Register(LoaderSignature);
+                _parent = parent;
+                _anomalyDict = new Dictionary<long, Double[]>();
+            }
+
+            public SrCnnEntireModeler(IHostEnvironment env, ModelLoadContext ctx)
+            {
+                Contracts.CheckValue(env, nameof(env));
+                _host = env.Register(LoaderSignature);
+
+                _anomalyDict = new Dictionary<long, Double[]>();
+
+                var length = ctx.Reader.ReadInt32();
+                _host.CheckDecode(length > 0);
+
+                var count = ctx.Reader.ReadInt32();
+                _host.CheckDecode(count >= 0);
+
+                if (count == 0)
+                    return;
+
+                List<long> trainingTimestamp = ctx.Reader.ReadLongArray(count).ToList();
+                List<double> valList = ctx.Reader.ReadDoubleArray(length * count).ToList();
+                List<Double[]> batchResult = new List<double[]>();
+                for (int i = 0; i < count; ++i)
+                {
+                    batchResult.Add(valList.GetRange(i * length, length).ToArray());
+                }
+                _anomalyDict = trainingTimestamp.Zip(batchResult, (k, v) => new { k, v }).ToDictionary(x => x.k, x => x.v);
+            }
+
+            void ICanSaveModel.Save(ModelSaveContext ctx) => SaveModel(ctx);
+
+            private void SaveModel(ModelSaveContext ctx)
+            {
+                _host.CheckValue(ctx, nameof(ctx));
+                ctx.CheckAtModel();
+                ctx.SetVersionInfo(GetVersionInfo());
+
+                ctx.Writer.Write((Int32)_parent.OutputLength);
+                ctx.Writer.Write((Int32)_anomalyDict.Count);
+
+                List<long> keyList = new List<long>();
+                List<double> valList = new List<double>();
+                foreach(var pair in _anomalyDict)
+                {
+                    keyList.Add(pair.Key);
+                    valList.AddRange(pair.Value);
+                }
+                ctx.Writer.WriteLongStream(keyList);
+                ctx.Writer.WriteDoubleStream(valList);
+            }
+
+            public void Train(IDataView data) => Train(new RoleMappedData(data, null, _parent.InputColumnName));
+
+            public void Train(RoleMappedData data)
```

Comment:
>RoleMappedData [](start = 30, length = 14)

There is no need for this overload, you can train directly on the `IDataView` since you know which column you are using as input (`_parent.InputColumnName`).

Reply:
> > RoleMappedData [](start = 30, length = 14)
> 
> There is no need for this overload, you can train directly on the `IDataView` since you know which column you are using as input (`_parent.InputColumnName`).

Removed in the new static class.